File: D.S.A/arrays/easy/L-rotate-byd-d.js
function LrotateByD(arr,k){
    let n=arr.length;

    let d=k%n;

    if(d===0){
        return arr;
    }
    
    arrrverse(0,d-1,arr);
    arrrverse(d,n-1,arr);

    arr.reverse();
}

function arrrverse(l,r,arr){
    while(l<r){
        [arr[l],arr[r]]=[arr[r],arr[l]];
        l++;
        r--;
    }
}

let arr=[1,2,3,4,5,6,7];
LrotateByD(arr,6);
console.log(arr);

File: Java/dsa/arrays/easy/arrays/LRotateByDD.java
public class LRotateByDD {
    public static void leftRotateByD(int[] arr, int k) {
        int n = arr.length;
        int d = k % n;
        if (d == 0) {
            return;
        }
        reverse(arr, 0, d - 1);
        reverse(arr, d, n - 1);
        reverse(arr, 0, n - 1);
    }

    private static void reverse(int[] arr, int l, int r) {
        while (l < r) {
            int temp = arr[l];
            arr[l] = arr[r];
            arr[r] = temp;
            l++;
            r--;
        }
    }
}
